### PR TITLE
Various small simplifications

### PR DESCRIPTION
**Pull request by anntzer**
## PR Summary

- more check_in_list
- remove unused import
- more literal sets
- more unpacking and star-unpacking

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by QuLogic** [comment]
Conflicts...

**Comment by anntzer** [comment]
rebased

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -2734,16 +2734,13 @@ def stem(self, *args, linefmt=None, markerfmt=None, basefmt=None, bottom=0,
             raise TypeError('stem expected between 1 and 5 positional '
                             'arguments, got {}'.format(args))
 
-        y = np.asarray(args[0])
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: Not sure if you need this `asarray` bit? I guess it's covered by `convert_xunits`?

Comment (anntzer) [reply lib/matplotlib/axes/_axes.py]: Yes; in fact this change means that stem now correctly handles units on y-data (whereas they would be previously dropped by the call to asarray).  Not an intentional change though :)

**Pull request merged**